### PR TITLE
Task/ad 660 custom selected item template

### PR DESCRIPTION
**Pull request by ericpitcher**

### CODE REVIEW

File: src/bim-select.component.js [hunk 1]
```diff
@@ -210,10 +227,16 @@ function BimSelectController(
     $ctrl.activateHandler = function(event) {
         event && event.stopPropagation();
         $ctrl.inputValue = '';
+        isFocused = true;
+        setSelected(null); // Fixes delayed selected item change glitch.
```
Comment (tregusti) [inline src/bim-select.component.js]: Please explain this better for posterity. What glitch etc.

File: package.json [hunk 2]
```diff
@@ -13,7 +13,8 @@
     "build:dev": "webpack --config webpack.dev.js",
     "build:prod": "webpack --config webpack.prod.js",
     "changelog": "node ./tools/bower-sync.js && conventional-changelog -p angular -i CHANGELOG.md -s",
-    "version": "conventional-recommended-bump -p angular"
+    "version": "conventional-recommended-bump -p angular",
+    "dev": "webpack-dev-server --config webpack.dev.js"
```
[the remark this answers is not in the document]
Comment (ericpitcher) [reply package.json]: Added webpack dev server to package.json and pointed content base to examples

File: src/bim-select.component.js (hunk 1 above)
Comment (tregusti) [reply src/bim-select.component.js]: Also, there is no test covering this addition.

File: src/bim-select.component.js [hunk 3]
```diff
@@ -182,6 +194,11 @@ function BimSelectController(
         $ctrl.internalItemTemplateUrl = $ctrl.itemTemplateUrl ||
             bimSelectConfig.itemTemplateUrl ||
             defaultItemTemplateUrl;
+        $ctrl.internalSelectedItemTemplateUrl = $ctrl.selectedItemTemplateUrl ||
+            $ctrl.itemTemplateUrl ||
+            bimSelectConfig.selectedItemTemplateUrl ||
+            bimSelectConfig.itemTemplateUrl ||
+            defaultSelectedItemTemplateUrl;
```
[the remark this answers is not in the document]
Comment (ericpitcher) [reply src/bim-select.component.js]: Fixed, provider takes precedence now

File: package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (tregusti) [reply package.json]: Works well!

File: src/bim-select.component.js (hunk 1 above)
Comment (ericpitcher) [reply src/bim-select.component.js]: Fixed comment in code

File: src/bim-select.less [hunk 4]
```diff
@@ -1,6 +1,13 @@
 bim-select {
     display: block;
-
+    .bim-select-input {
+        opacity: 0 !important;
+        border-top-left-radius: 4px !important;
+        border-bottom-left-radius: 4px !important;
```
Comment (tregusti) [inline src/bim-select.less]: Why `!important`? Just add comment about why if it's needed. This way we know when it's safe to remove it again. At least if no tests cover it.

Comment (ericpitcher) [reply src/bim-select.less]: Fixed

File: src/bim-select.component.js [hunk 5]
```diff
@@ -158,7 +167,10 @@ function BimSelectController(
 ) {
     const $ctrl = this;
     const defaultItemTemplateUrl = itemTemplateUrl;
+    const defaultSelectedItemTemplateUrl = selectedItemTemplateUrl;
     const ul = $element[0].querySelector('ul');
+    let isFocused = false;
+    $scope.item = null;
```
[the remark this answers is not in the document]
Comment (ericpitcher) [reply src/bim-select.component.js]: Fixed

**Pull request merged**